### ISSUE TITLE
Refactor and fix configuration-reference for AKS

### ISSUE THREAD

**Issue by knrt10** [kinvolk/lokomotive#1302]
**Description**

The configuration reference is a bit non-informative. There are places where the first-time users won't have information about what to do. I will update this issue while trying to create cluster

- Some fields in attribute reference are marked as not required but still required for example `client_id`.
- Taints value is not correct, and the cluster fails with the following output

```bash
Error: creating Managed Kubernetes Cluster "knrt10" (Resource Group "knrt10-dev"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="The value of parameter agentPoolProfile.nodeTaints is invalid. Error details: unknown taint spec: node-role.kubernetes.io/master=NoSchedule. For more details, please visit https://aka.ms/node-taints. Please see https://aka.ms/aks-naming-rules for more details." Target="agentPoolProfile.nodeTaints
```

**Impact**

Unable to create AKS cluster

**Environment and steps to reproduce**

1. Go to https://github.com/kinvolk/lokomotive/blob/master/docs/configuration-reference/platforms/aks.md

**Expected behavior**
Should be able to easily create the cluster.

**Comment by invidian** [comment]
>Error: creating Managed Kubernetes Cluster "knrt10" (Resource Group "knrt10-dev"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="The value of parameter agentPoolProfile.nodeTaints is invalid. Error details: unknown taint spec: node-role.kubernetes.io/master=NoSchedule. For more details, please visit https://aka.ms/node-taints. Please see https://aka.ms/aks-naming-rules for more details." Target="agentPoolProfile.nodeTaints

This is correct. Taints use `:` and `=`, not only `=` like labels. `node-role.kubernetes.io/master=NoSchedule` is not a correct taint. but why do we apply such taint in the first place? AKS manages it's own controller nodes.

>Some fields in attribute reference are marked as not required but still required for example client_id.

Well, `client_id` can also be provided using `LOKOMOTIVE_AKS_CLIENT_ID` environment variable, so technically it is not required in the configuration. Our documentation cannot really define such condition unless we add environment variable column or something :/
